flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinator.java

```diff
@@ -166,53 +171,57 @@ public SourceCoordinator(
 
     @VisibleForTesting
     void announceCombinedWatermark() {
-        Set<Integer> subTaskIds = combinedWatermark.keySet();
-        if (subTaskIds.isEmpty()) {
-            LOG.debug(
-                    "Skip distributing maxAllowedWatermark of group={} for source {} - no subtasks.",
-                    watermarkAlignmentParams.getWatermarkGroup(),
-                    operatorName);
-        }
-        checkState(
-                watermarkAlignmentParams != WatermarkAlignmentParams.WATERMARK_ALIGNMENT_DISABLED);
-
-        Watermark globalCombinedWatermark =
-                coordinatorStore.apply(
+        try (MdcUtils.MdcCloseable mdcCloseable =
+                MdcUtils.withContext(MdcUtils.asContextData(jobID))) {
```

Comment:
Have you tried to add MDC to the executor that runs this function instead of wrapping here?
It looks like it would be enough to wrap `coordinatorExecutor` in `SourceCoordinatorContext`.

That would cover more cases (i.e. if new methods are added) and less invasive and verbose.